sogs/routes/dm.py

```diff
@@ -112,3 +110,25 @@ def send_inbox(sid):
     with db.transaction():
         msg = Message(data=utils.decode_base64(message), recip=recip_user, sender=g.user)
     return jsonify(_serialize_message(msg, include_message=False)), http.CREATED
+
+
+@dm.delete("/inbox")
+@auth.blind_user_required
+def delete_inbox_items():
+    """
+    Deletes the user's sent and recieved messages.
+    # Return value
+
+    204 No Content - On successful removal of messages.
+    304 Not Modified - If there were no messages to remove.
+
+    """
+
+    num_deleted = 0
+    with db.transaction():
+        num_deleted += Message.delete_all(recip=g.user)
+        num_deleted += Message.delete_all(sender=g.user)
```

Comment:
I'm don't think we should allow deleting sent messages.  Sent messages are the property of the recipient in regular Session message model (at best you can send another deletion meta-request follow-up) and I'm thinking that should be mirrored here.

Reply:
Additionally, this request is for `DELETE /inbox`, so deleting everything in the outbox as well doesn't seem right.

(But as to my previous point: I don't think we should have a `DELETE /outbox`).